test/functional/shada/history_spec.lua

```diff
@@ -173,158 +171,57 @@ describe('ShaDa support code', function()
-    nvim_command('substitute/./«/ge')
-    set_additional_cmd('set encoding=latin1')
-    nvim_command('qall')
-    reset()
-    funcs.setline('.', {'.'})
-    nvim_command('&')
-    eq('\171', funcs.getline('.'))
-  end)
-
-  it('dumps and loads substitute pattern correctly when &encoding is not UTF-8',
+  it('dumps and loads substitute pattern with UTF-8 characters',
   function()
-    set_additional_cmd('set encoding=latin1')
-    reset()
-    -- \171 is U+00AB LEFT-POINTING DOUBLE ANGLE QUOTATION MARK in latin1
-    nvim_command('substitute/\171/./ge')
+    nvim_command('substitute/«/./ge')
     nvim_command('qall!')
     reset()
-    funcs.setline('.', {'\171«'})
-    nvim_command('&')
-    eq('.«', funcs.getline('.'))
-  end)
-
-  it('dumps&loads s/pattern correctly when &encoding /= UTF-8 when dumping',
-  function()
-    set_additional_cmd('set encoding=latin1')
-    reset()
-    -- \171 is U+00AB LEFT-POINTING DOUBLE ANGLE QUOTATION MARK in latin1
-    nvim_command('substitute/\171/./ge')
-    set_additional_cmd('')
-    nvim_command('qall')
-    reset()
     funcs.setline('.', {'«\171'})
     nvim_command('&')
+    --nvim_command('substitute/«/./ge')
     eq('.\171', funcs.getline('.'))
   end)
 
-  it('dumps&loads s/pattern correctly when &encoding /= UTF-8 when loading',
-  function()
-    -- \171 is U+00AB LEFT-POINTING DOUBLE ANGLE QUOTATION MARK in latin1
-    nvim_command('substitute/«/./ge')
-    set_additional_cmd('set encoding=latin1')
-    nvim_command('qall')
-    reset()
-    funcs.setline('.', {'\171«'})
-    nvim_command('&')
-    eq('.«', funcs.getline('.'))
-  end)
-
-  it('dumps and loads search pattern correctly when &encoding is not UTF-8',
+  it('dumps and loads search pattern with UTF-8 characters',
   function()
-    set_additional_cmd('set encoding=latin1')
-    reset()
-    -- \171 is U+00AB LEFT-POINTING DOUBLE ANGLE QUOTATION MARK in latin1
-    nvim_command('silent! /\171/')
+    nvim_command('silent! /«/')
     nvim_command('set shada+=/0')
     nvim_command('qall!')
     reset()
     funcs.setline('.', {'\171«'})
     nvim_command('~&')
-    eq('«', funcs.getline('.'))
-    eq('', funcs.histget('/', -1))
-  end)
-
-  it('dumps&loads /pattern correctly when &encoding /= UTF-8 when dumping',
-  function()
-    set_additional_cmd('set encoding=latin1')
-    reset()
-    -- \171 is U+00AB LEFT-POINTING DOUBLE ANGLE QUOTATION MARK in latin1
-    nvim_command('silent! /\171/')
-    nvim_command('set shada+=/0')
-    set_additional_cmd('')
-    nvim_command('qall')
-    reset()
-    funcs.setline('.', {'«\171'})
-    nvim_command('~&')
     eq('\171', funcs.getline('.'))
     eq('', funcs.histget('/', -1))
   end)
 
-  it('dumps&loads /pattern correctly when &encoding /= UTF-8 when loading',
+  it('dumps and loads search pattern with 8-bit characters',
   function()
-    -- \171 is U+00AB LEFT-POINTING DOUBLE ANGLE QUOTATION MARK in latin1
-    nvim_command('silent! /«/')
+    nvim_command('silent! /\171/')
```

Comment:
perhaps the `-- \171 is U+00AB LEFT-POINTING ...` comment should be kept in this case